ServiceApp/server/src/utils/credentialHelper.ts

```diff
@@ -29,7 +31,78 @@ export interface IUser {
     address: string;
 }
 
-export function createNewUser(name: string = '', role: string = '', location: string = ''): Promise <IUser> {
+export function createNewUserC2(name: string = '', role: string = '', location: string = ''): Promise<IUser> {
+    return new Promise<IUser>(async (resolve, reject) => {
+        const clientConfig = {
+            network: "main",
+            node: "https://chrysalis-nodes.iota.org:443",
+        }
+        try {
+
+            // Create a new DID Document with the KeyPair as the default authentication method
+            //@ts-ignore
+            const { doc, key } = new Document(KeyType.Ed25519, clientConfig.network);
+
+            const keyId = "#key";
+
+            const privateKey = key.secret;
+            const publicKey = key.public;
+
+            //TODO: For now static
+            const serviceEndpoint = "iota1qpw6k49dedaxrt854rau02talgfshgt0jlm5w8x9nk5ts6f5x5m759nh2ml" //TODO: Generate address here
+
+            //Add a new ServiceEndpoint
+            //TODO: nameFragment does not exist anymore + concept of serviceEndpoints=address still valid?
+            const service: any = {
+                "id": doc.id + "#tanglecom", //TODO: Is this how I would now insert a nameFragment?
+                "type": "TangleCommunicationAddress",
+                "serviceEndpoint": serviceEndpoint
+            };
+
+            // doc.insertService(Service.fromJSON(service));
+
+            //Needed, as serviceEndpoint currently only works with URIs and not with IOTA addresses. Scheme is otherwise the same (attribute is only called serviceDummy instead of service)
+            const serviceDummy = [service]
+            const docWithService = Document.fromJSON({
+                ...doc.toJSON(),
+                serviceDummy
+            });
+
+
+            docWithService.sign(key);
+
+            if (!docWithService.verify()) {
+                reject('Created DID is not valid!');
+            }
+
+
+            // Create a default client configuration from the parent config network.
+            const config = Config.fromNetwork(Network.mainnet());
+
+            // Create a client instance to publish messages to the Tangle.
+            const client = Client.fromConfig(config);
+
+            // Publish the Identity to the IOTA Network, this may take a few seconds to complete Proof-of-Work.
+            const messageId = await client.publishDocument(docWithService.toJSON());
+
+            //TODO: "keyId": was previously "keys-1", now recommended #_sign-1 -Default is however "#key"
+            await writeData('didC2', { messageId, id: docWithService.id.toString(), privateKey, publicKey, keyId });
+
+            const docId = docWithService.id.toString();
+
+            const user: IUser = { id: docId, name, role, location, address: serviceEndpoint };
+
+            await writeData('user', user);
```

Comment:
do you also store the generated `key` somewhere?

Reply:
Yes, I currently store `key.secret` and `key.public ` in the db-table-`didC2`